pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -876,7 +876,7 @@ func (m *mockMonitor) ObserveMetricComputationResult(action monitor.ActionLabel,
 
 // waitUntilRecorded waits for the HPA controller to reconcile at least once.
 func (m *mockMonitor) waitUntilRecorded(t *testing.T) {
-	if err := wait.Poll(20*time.Millisecond, 100*time.Millisecond, func() (done bool, err error) {
+	if err := wait.PollUntilContextTimeout(context.Background(), 20*time.Millisecond, 100*time.Millisecond, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
👍 on replacing deprecated function

I know this is just a test but we could dispatch the `ctx` object instantiated in `runTestWithController` down to `verifyResults` and then to `verifyRecordedMetric` and then finally into `waitUntilRecorded`. Then we'd have a more hygienic context to pass into `wait.PollUntilContextTimeout`.

Reply:
Yup, good idea!

Done in 42a5e5f4255